src/lib/utils/hooks/SynapseAPI/useEntity.ts

```diff
@@ -12,18 +14,56 @@ import {
 import { SynapseClient } from '../..'
 import { SynapseClientError } from '../../SynapseClient'
 import { useSynapseContext } from '../../SynapseContext'
-import { EntityJson } from '../../synapseTypes'
+import { EntityJson, entityJsonKeys, EntityJsonValue } from '../../synapseTypes'
 
+export function getStandardEntityFields(json: EntityJson): EntityJson {
+  return pick(json, entityJsonKeys) as EntityJson
```

Comment:
I suppose this is where entityJsonKeys is the union of all possible entity object field names.  I think this is fine, but should we be worried about the edge cases (like a FileEntity annotated with repositoryName)?

Reply:
It would cause a problem if we conditionally render content using presence of the property, rather than the concrete type (which I think we do quite a bit).

[We do show an error in the new editor if a user tries to save such a property](https://github.com/Sage-Bionetworks/Synapse-React-Client/blob/ca0ebdbdcc59d61781999db21a5eeb8d176ba0d0/src/lib/containers/entity/annotations/SchemaDrivenAnnotationEditor.tsx#L242), but they can still click through the error. The backend doesn't block it.

Reply:
We would also only see this issue if we use the JSON service, which (for now), we only use for getting/setting annotations